aces/idt/generators/prosumer_camera.py

```diff
@@ -0,0 +1,602 @@
+    -   :attr:`~aces.idt.IDTGeneratorProsumerCamera.samples_decoded`
+    -   :attr:`~aces.idt.IDTGeneratorProsumerCamera.samples_weighted`
+    -   :attr:`~aces.idt.IDTGeneratorProsumerCamera.LUT_unfiltered`
+    -   :attr:`~aces.idt.IDTGeneratorProsumerCamera.LUT_filtered`
+    -   :attr:`~aces.idt.IDTGeneratorProsumerCamera.LUT_decoding`
+    -   :attr:`~aces.idt.IDTGeneratorProsumerCamera.M`
+    -   :attr:`~aces.idt.IDTGeneratorProsumerCamera.RGB_w`
+    -   :attr:`~aces.idt.IDTGeneratorProsumerCamera.k`
+
+    Methods
+    -------
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.__str__`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.from_specification`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.sample`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.sort`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.generate_LUT`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.filter_LUT`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.decode`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.optimise`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.to_clf`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.zip`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.png_colour_checker_segmentation`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.png_grey_card_sampling`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.png_measured_camera_samples`
+    -   :meth:`~aces.idt.IDTGeneratorProsumerCamera.png_extrapolated_camera_samples`
+    """
+
+    generator_name = "IDTGeneratorProsumerCamera"
+
+    def __init__(self, application):
+        super().__init__(application)
+
+        self._lut_blending_edge_left = None
+        self._lut_blending_edge_right = None
+
+        self._samples_decoded = None
+        self._samples_weighted = None
+
+    @property
+    def samples_decoded(self):
+        """
+        Getter property for the samples of the camera decoded by applying the
+        filtered *LUT*.
+
+        Returns
+        -------
+        :class:`NDArray` or None
+            Samples of the camera decoded by applying the filtered *LUT*.
+        """
+
+        return self._samples_decoded
+
+    @property
+    def samples_weighted(self):
+        """
+        Getter property for the decoded samples of the camera weighted across
+        multiple exposures.
+
+        Returns
+        -------
+        :class:`NDArray` or None
+            Samples of the decoded samples of the camera weighted across
+            multiple exposures.
+        """
+
+        return self._samples_weighted
+
+    def __str__(self):
+        """
+        Return a formatted string representation of the *IDT* generator.
+
+        Returns
+        -------
+        :class:`str`
+            Formatted string representation.
+        """
+        # TODO Can't do this can't have the __str__ method be dependent on an execution
+        #  state of the instance
+        #  _samples_analysis may not be populated yet
```

Comment:
`_samples_analysis may not be populated yet`:

Which is actually an indication that sampling has not occurred! Unsure why `__str__` method should not be dependent on an execution state of the instance, it's purpose is to return a string representation of the instance, so it is natural that as a function of instance life cycle, this representation changes.

Reply:
Totally get the need to see the status of the object

my point was I currently get a stack trace when I try and print this object out when the sampling has not been run

Do we want that? 

__str__ should always give you something usable.

This feels dirty to me to have it stack trace.

Id vote for at least handling it and simply saying No Samples or something

Reply:
Ok so ive updated this so that it doesnt error also added this to the project settings and also so that we retain the standard python object info for debugging

`<aces.idt.generators.prosumer_camera.IDTGeneratorProsumerCamera object at 0x334358710>

Baseline Exposure : 0
Samples Analysis  : None
LUT Decoding      : None
M                 : None
RGB_w             : None
k                 : None
<aces.idt.framework.project_settings.IDTProjectSettings object at 0x16ac7dc50>

Schema_Version             : 0.1.0
Aces_Transform_Id          : 
Aces_User_Name             : 
Camera_Make                : 
Camera_Model               : 
Iso                        : 800
Temperature                : 6000
Additional_Camera_Settings : 
Lighting_Setup_Description : 
Debayering_Platform        : 
Debayering_Settings        : 
Encoding_Colourspace       : 
Rgb_Display_Colourspace    : sRGB
Cat                        : CAT02
Optimisation_Space         : Oklab
Illuminant_Interpolator    : Linear
Decoding_Method            : Median
Ev_Range                   : [-1.0, 0.0, 1.0]
Grey_Card_Reference        : [0.18, 0.18, 0.18]
Lut_Size                   : 1024
Lut_Smoothing              : 32
Data                       : {'colour_checker': {}, 'grey_card': {}}
Working_Directory          : 
Cleanup                    : False
Reference_Colour_Checker   : ISO 17321-1
Illuminant                 : D60
Sigma                      : 16
File_Type                  : 
Ev_Weights                 : []
Optimization_Kwargs        : {}
`